src/EditorFeatures/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeGeneration
+{
+    [UseExportProvider]
+    public class SyntaxGeneratorTests
+    {
+        [Fact]
+        public async Task TestNameOfBindsWithoutErrors()
```

Comment:
💡 An additional test for a speculative semantic model would be good as well. This is a more likely scenario for analyzer authors to hit.

Reply:
will do.